src/adagrams.js

```diff
@@ -1,7 +1,151 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
+class Adagrams {
+  static buildPool() {
+    let pool = [];
+      const poolCount = {
+        A: 9,
+        B: 2,
+        C: 2,
+        D: 4,
+        E: 12,
+        F: 2,
+        G: 3,
+        H: 2,
+        I: 9,
+        J: 1,
+        K: 1,
+        L: 4,
+        M: 2,
+        N: 6,
+        O: 8,
+        P: 2,
+        Q: 1,
+        R: 6,
+        S: 4,
+        T: 6,
+        U: 4,
+        V: 2,
+        W: 2,
+        X: 1,
+        Y: 2,
+        Z: 1,
+      };
+      
+      for (const letter in poolCount) {
+        let count = poolCount[letter];
+        while (count > 0) {
+            pool.push(letter);
+            count--;
+        }
+      }
+    return pool;
+  }
+
+  static drawLetters() {
+    const __ = require('underscore')
+    const pool = this.buildPool();
+    return __.sample(pool, 10);
```

Comment:
I would have loved a comment explaining what underscore does, because it seems to be doing a lot of the heavy lifting here. While I think this resourcefulness is going to server you *really well* as an engineer, one of the goals of the project is to make sure that you are comfortable crunching out some tricky JS by hand.